escape.go

```diff
@@ -210,140 +204,122 @@ func (ei *escapeInterpreter) parseOne(ch rune) (isEscape bool, err error) {
 	return false, nil
 }
 
-// outputNormal provides 8 different colors:
-//   black, red, green, yellow, blue, magenta, cyan, white
-func (ei *escapeInterpreter) outputNormal() error {
-	for _, param := range ei.csiParam {
-		p, err := strconv.Atoi(param)
+func (ei *escapeInterpreter) outputCSI() error {
+	n := len(ei.csiParam)
+	for i := 0; i < n; {
+		p, err := strconv.Atoi(ei.csiParam[i])
 		if err != nil {
 			return errCSIParseError
 		}
 
+		skip := 1
 		switch {
-		case p >= 30 && p <= 37:
-			ei.curFgColor = Get256Color(int32(p) - 30)
-		case p == 39:
+		case p == 0: // reset style and color
 			ei.curFgColor = ColorDefault
-		case p >= 40 && p <= 47:
-			ei.curBgColor = Get256Color(int32(p) - 40)
-		case p == 49:
 			ei.curBgColor = ColorDefault
-		case p == 0:
-			ei.curFgColor = ColorDefault
-			ei.curBgColor = ColorDefault
-		case p >= 21 && p <= 29:
+		case p >= 1 && p <= 9: // set style
+			ei.curFgColor |= getFontEffect(p)
+		case p >= 21 && p <= 29: // reset style
 			ei.curFgColor &= ^getFontEffect(p - 20)
+		case p >= 30 && p <= 37: // set foreground color
+			ei.curFgColor &= AttrStyleBits
+			ei.curFgColor |= Get256Color(int32(p) - 30)
+		case p == setForegroundColor: // set foreground color
```

Comment:
Both this case and the above case say 'set foreground color'. I'm not clear on how they're different though.